src/main/java/io/getunleash/repository/FeatureRepositoryImpl.java

```diff
@@ -0,0 +1,219 @@
+            Runnable updateFeatures = updateFeatures(this.eventDispatcher::dispatch);
+            if (unleashConfig.getFetchTogglesInterval() > 0) {
+                executor.setInterval(updateFeatures, 0, unleashConfig.getFetchTogglesInterval());
+            } else {
+                executor.scheduleOnce(updateFeatures);
+            }
+        }
+    }
+
+    private Runnable updateFeatures(final Consumer<UnleashException> handler) {
+        return () -> {
+            if (throttler.performAction()) {
+                try {
+                    ClientFeaturesResponse response = featureFetcher.fetchFeatures();
+                    eventDispatcher.dispatch(response);
+                    if (response.getStatus() == ClientFeaturesResponse.Status.CHANGED) {
+                        String clientFeatures = response.getClientFeatures().get();
+
+                        this.engine.takeState(clientFeatures);
+                        featureBackupHandler.write(clientFeatures);
+                    } else if (response.getStatus() == ClientFeaturesResponse.Status.UNAVAILABLE) {
+                        if (!ready && unleashConfig.isSynchronousFetchOnInitialisation()) {
+                            throw new UnleashException(
+                                    String.format(
+                                            "Could not initialize Unleash, got response code %d",
+                                            response.getHttpStatusCode()),
+                                    null);
+                        }
+                        if (ready) {
+                            throttler.handleHttpErrorCodes(response.getHttpStatusCode());
+                        }
+                        return;
+                    }
+                    throttler.decrementFailureCountAndResetSkips();
+                    if (!ready) {
+                        eventDispatcher.dispatch(new UnleashReady());
+                        ready = true;
+                    }
+                } catch (UnleashException e) {
+                    handler.accept(e);
+                } catch (YggdrasilInvalidInputException | YggdrasilError e) {
+                    handler.accept(new UnleashException("Error when fetching features", e));
+                }
+            } else {
+                throttler.skipped(); // We didn't do anything this iteration, just reduce the count
+            }
+        };
+    }
+
+    public Integer getFailures() {
+        return this.throttler.getFailures();
+    }
+
+    public Integer getSkips() {
+        return this.throttler.getSkips();
+    }
+
+    @Override
+    public Boolean isEnabled(String toggleName, UnleashContext context) {
+        try {
+            return this.engine.isEnabled(toggleName, YggdrasilAdapters.adapt(context));
+        } catch (YggdrasilInvalidInputException | YggdrasilError e) {
+            LOGGER.error("Error when checking feature toggle {}", toggleName, e);
+            return null;
```

Comment:
Do we want to return null or just false? I know this would be a major issue... but between returning null and just throwing I believe I err towards throwing, cause a NullPointerException would be confusing, but I believe the option I like the most is:

```suggestion
            return false; // if we had the default value for the feature here, that'd be even better
```

Reply:
The empty response here is intentional. SDKs should handle missing toggles vs false toggles differently and that's in how the fallback function is executed (and for other SDKs how metrics are calculated). This lets us signal to the layer above that the toggle isn't here without having to leak the fallback handling into this layer

Could swap this for an Optional<boolean> if this bothers you though?

Reply:
I think  it's fine as long as FeatureRepository is an internal interface (or meant to be internal). If it's exposed as something a client can use then I think optional is better, but in this case I understand the point and I think it's fine to return null